firebase-vscode/common/messaging/broker.ts

```diff
@@ -0,0 +1,87 @@
+import { MessageParamsMap } from "./protocol";
+import { Listener, Message, MessageListeners } from "./types";
+import { Webview } from "vscode";
+
+const isObject = (val: any): boolean => typeof val === "object" && val !== null;
+
+type Receiver = {} | Webview;
+
+export abstract class Broker<
```

Comment:
I messed around trying to tighten up the types, let me know if this is a good idea or just a big mess.

Reply:
We can always loosen them later IMO - best to start strict than the other way around